test/controllers/products_controller_test.rb

```diff
@@ -84,5 +98,79 @@
         must_respond_with :not_found
       end
     end
+
+    describe "new action" do
+      it "redirects" do
+        get new_wizard_product_path(wizard1.id)
+        must_respond_with :redirect
+      end
+    end
+
+    describe "create action" do
+      it "redirects" do
+        post wizard_products_path(wizard1.id), params: new_product_params
+        must_respond_with :redirect
+      end
+    end
+  end
+
+  describe "Logged In Users Only" do
+    before do
+      perform_login(wizard1)
+    end
+    describe "new action" do
+      it "succeeds if user is the Wizard given in params and Wizard ID matches logged in user's ID" do
+        get new_wizard_product_path(wizard1.id)
+        must_respond_with :success
+      end
+
+      it "redirects if correct wizard is not logged in" do
+        different_wizard = wizard_no_products
+        get new_wizard_product_path(different_wizard.id)
+        must_respond_with :redirect
+      end
+    end
+
+    describe "create action" do
+      it "redirects if correct wizard is not logged in" do
+        different_wizard = wizard_no_products
+        post wizard_products_path(different_wizard.id), params: new_product_params
+        must_respond_with :redirect
+      end
+
+      it "creates a product given valid product data" do
+        expect {
+          post wizard_products_path(wizard1.id), params: new_product_params
+        }.must_change "Product.count", 1
+
+        must_respond_with :redirect
+      end
+
+      it "renders bad_request and does not update the DB for bogus data" do
+        bad_new_product_params = {
+          product: {
+            name: nil,
+            description: "Amulet that also qualifies as an artifact",
+            stock: 0,
+            # photo_url,
```

Comment:
recommend removing commented out lines. Or we can do a cleanup before submitting everything.